fem/fespace.cpp

```diff
@@ -2881,238 +3002,1215 @@ void ElementRestriction::MultTranspose(const Vector& x, Vector& y) const
    });
 }
 
-
```

Comment:
There is a lot of changes in this already huge file.
Some of the them collides with existing code, even GitHub is confused.
Open question: Could we start splitting this `Face` additions to a new files?

Reply:
@tzanio @v-dobrev Is it okay if I move all the code related to restrictions to `fem/restriction.hpp` and 'fem/restriction.cpp`?

Reply:
If it is a separate class, yes. Just avoid having implementations of the same class in multiple files.

Reply:
Sounds good to me.

Reply:
done.